sdks/csharp/sdk/Alpha.cs

```diff
@@ -210,6 +211,380 @@ public async Task<List<string>> GetConnectedPlayersAsync()
+        /// and the string value. Search is case-sensitive.
+        /// Will error if the key was not predefined in the GameServer resource on creation.
+        /// </summary>
+        /// <returns>True if the value is found in the List</returns>
+        public async Task<bool> ListContainsAsync(string key, string value)
+        {
+            try
+            {
+                var request = new GetListRequest()
+                {
+                    Name = key,
+                };
+                var list = await client.GetListAsync(request,
+                  deadline: DateTime.UtcNow.AddSeconds(RequestTimeoutSec), cancellationToken: ctoken);
+                foreach (string val in list.Values)
+                {
+                    if (val == value)
+                    {
+                        return true;
+                    };
+                }
+                return false;
+            }
+            catch (RpcException ex)
+            {
+                LogError(ex, $"Unable to invoke ListContains({key}, {value}).");
+                throw;
+            }
+        }
+
+        /// <summary>
+        /// GetListLengthAsync returns the length of the Values list for a List, given the List's key.
+        /// Will error if the key was not predefined in the GameServer resource on creation.
+        /// </summary>
+        /// <returns>The length of List's values array</returns>
+        public async Task<int> GetListLengthAsync(string key)
+        {
+            try
+            {
+                var request = new GetListRequest()
+                {
+                    Name = key,
+                };
+                var list = await client.GetListAsync(request,
+                  deadline: DateTime.UtcNow.AddSeconds(RequestTimeoutSec), cancellationToken: ctoken);
+                return list.Values.Count;
+            }
+            catch (RpcException ex)
+            {
+                LogError(ex, $"Unable to invoke GetListLength({key}).");
+                throw;
+            }
+        }
+
+        /// <summary>
+        /// GetListValuesAsync returns the Values for a List, given the List's key (name).
+        /// Will error if the key was not predefined in the GameServer resource on creation.
+        /// </summary>
+        /// <returns>The List's values array</returns>
+        public async Task<List<string>> GetListValuesAsync(string key)
+        {
+            try
+            {
+                var request = new GetListRequest()
+                {
+                    Name = key,
+                };
+                var list = await client.GetListAsync(request,
+                  deadline: DateTime.UtcNow.AddSeconds(RequestTimeoutSec), cancellationToken: ctoken);
+                var values = new string[list.Values.Count];
+                list.Values.CopyTo(values, 0);
+                return values.ToList();

```

Comment:
Since `Values` is already a collection (a repeated field), you should be able to use `.ToList()` on it
```suggestion
                return list.Values.ToList();

```

Reply:
Thanks! I'm new to C#, so this is helpful.

Reply:
Alternatively, if this method was changed to return `Task<IReadOnlyList<string>>`, you could just `return list.Values;` because `Values` is a `repeated` field https://github.com/googleforgames/agones/blob/main/proto/sdk/alpha/alpha.proto#L276 and in the C# protobuf library, `RepeatedField<T>` implements `IReadOnlyList<T>` https://github.com/protocolbuffers/protobuf/blob/main/csharp/src/Google.Protobuf/Collections/RepeatedField.cs#L31